src/frontend/hooks/server/projects.ts

```diff
@@ -106,87 +39,30 @@ export const useOriginalVersionIdToDeviceId = (originalVersionId: string) => {
   });
 };
 
-export function useLeaveProject() {
-  const mapeoApi = useClientApi();
-
-  const queryClient = useQueryClient();
-
-  return useMutation({
-    mutationFn: (projectId: string) => {
-      return mapeoApi.leaveProject(projectId);
-    },
-    onSuccess: () => {
-      queryClient.invalidateQueries({
-        queryKey: [ALL_PROJECTS_KEY],
-      });
-    },
-  });
-}
-
-export function useImportProjectConfig() {
-  const queryClient = useQueryClient();
-  const {projectApi} = useActiveProject();
-
-  return useMutation({
-    mutationFn: (configPath: string) => {
-      return projectApi.importConfig({configPath});
-    },
-    onSuccess: () => {
-      return Promise.all([
-        queryClient.invalidateQueries({
-          queryKey: [FIELDS_KEY],
-        }),
-        queryClient.invalidateQueries({
-          queryKey: [ICONS_KEY],
-        }),
-        queryClient.invalidateQueries({
-          queryKey: [PROJECT_SETTINGS_KEY],
-        }),
-        queryClient.invalidateQueries({
-          queryKey: [PRESETS_KEY],
-        }),
-      ]);
-    },
-  });
-}
-
 export function useGetOwnRole() {
-  const {projectId, projectApi} = useActiveProject();
+  const {data: deviceInfo} = useOwnDeviceInfo();
+  const {projectId} = useActiveProject();
 
-  return useQuery({
-    queryKey: [THIS_USERS_ROLE_KEY, projectId],
-    queryFn: () => {
-      return projectApi.$getOwnRole();
-    },
+  const {data} = useSingleMember({
+    projectId: projectId,
+    deviceId: deviceInfo.deviceId,
   });
+  return data.role;
 }
 
 export function useGetRemoteArchives() {
-  const {projectId, projectApi} = useActiveProject();
-
-  return useQuery({
-    queryKey: [REMOTE_ARCHIVE, projectId, PROJECT_MEMBERS_KEY],
-    queryFn: async () => {
-      const members = await projectApi.$member.getMany();
-      const filteredMembers = members.filter(
-        member => member.deviceType === 'selfHostedServer',
-      );
-      return filteredMembers;
-    },
-  });
-}
-
-export function useAddRemoteArchive() {
-  const {projectApi} = useActiveProject();
-  const queryClient = useQueryClient();
-  return useMutation({
-    mutationFn: async (normalizedUrl: string) => {
-      return projectApi.$member.addServerPeer(normalizedUrl);
-    },
-    onSuccess: () => {
-      queryClient.invalidateQueries({queryKey: [REMOTE_ARCHIVE]});
-    },
-  });
+  const {projectId} = useActiveProject();
+  const {data: members, error, isRefetching} = useManyMembers({projectId});
+
+  const archives = useMemo(() => {
+    return members?.filter(m => m.deviceType === 'selfHostedServer') ?? [];
+  }, [members]);
+
+  return {
+    data: archives,
+    error,
+    isPending: isRefetching,
```

Comment:
would suggest using `isRefetching` as the name of the field. `isPending` is subtlety different than `isRefetching` in react query so could easily get a bit confusing if renamed.

```suggestion
    isRefetching,
```